script/universal/MultisigBuilder.sol

```diff
@@ -42,130 +182,493 @@ abstract contract MultisigBuilder is MultisigBase {
+        }
+        return (calls);
+    }
+
+    function _overrides(address[] memory _safes) internal view returns (Simulation.StateOverride[] memory) {
         Simulation.StateOverride[] memory simOverrides = _simulationOverrides();
-        Simulation.StateOverride[] memory overrides = new Simulation.StateOverride[](1 + simOverrides.length);
-        overrides[0] = _safeOverrides(_safe, msg.sender);
+        Simulation.StateOverride[] memory overrides =
+            new Simulation.StateOverride[](_safes.length + simOverrides.length);
+        for (uint256 i = 0; i < _safes.length; i++) {
+            address owner = i == 0 ? MULTICALL3_ADDRESS : address(0);
+            overrides[i] = _safeOverrides(_safes[i], owner);
+        }
         for (uint256 i = 0; i < simOverrides.length; i++) {
-            overrides[i + 1] = simOverrides[i];
+            overrides[i + _safes.length] = simOverrides[i];
         }
         return overrides;
     }
+
+    // The state change simulation can set the threshold, owner address and/or nonce.
+    // This allows simulation of the final transaction by overriding the threshold to 1.
+    // State changes reflected in the simulation as a result of these overrides will
+    // not be reflected in the prod execution.
+    function _safeOverrides(address _safe, address _owner)
+        internal
+        view
+        virtual
+        returns (Simulation.StateOverride memory)
+    {
+        uint256 _nonce = _getNonce(_safe);
+        if (_owner == address(0)) {
+            return Simulation.overrideSafeThresholdAndNonce(_safe, _nonce);
+        }
+        return Simulation.overrideSafeThresholdOwnerAndNonce(_safe, _owner, _nonce);
+    }
+
+    // Get the nonce to use for the given safe, for signing and simulations.
+    //
+    // If you override it, ensure that the behavior is correct for all contexts.
+    // As an example, if you are pre-signing a message that needs safe.nonce+1 (before
+    // safe.nonce is executed), you should explicitly set the nonce value with an env var.
+    // Overriding this method with safe.nonce+1 will cause issues upon execution because
+    // the transaction hash will differ from the one signed.
+    //
+    // The process for determining a nonce override is as follows:
+    //   1. We look for an env var of the name SAFE_NONCE_{UPPERCASE_SAFE_ADDRESS}. For example,
+    //      SAFE_NONCE_0X6DF4742A3C28790E63FE933F7D108FE9FCE51EA4.
+    //   2. If it exists, we use it as the nonce override for the safe.
+    //   3. If it does not exist, we do the same for the SAFE_NONCE env var.
+    //   4. Otherwise we fallback to the safe's current nonce (no override).
+    function _getNonce(address _safe) internal view virtual returns (uint256 nonce) {
+        uint256 safeNonce = IGnosisSafe(_safe).nonce();
+        nonce = safeNonce;
+
+        // first try SAFE_NONCE
+        try vm.envUint("SAFE_NONCE") {
+            nonce = vm.envUint("SAFE_NONCE");
+        } catch {}
```

Comment:
This is slightly different in that it now tries to load `SAFE_NONCE` for all levels of the safe, rather than just the non-nested version. Retained for backwards compatibility.